client.go

```diff
@@ -235,21 +235,37 @@ func (r *ErrorResponse) Error() string {
 		r.Response.Request.Method, r.Response.Request.URL, r.Response.StatusCode, r.TransferError, r.Message)
 }
 
-func CheckResponse(r *http.Response) error {
+func CheckResponse(r *http.Response, responseBody interface{}) (bool, error) {
```

Comment:
```suggestion
func CheckResponse(r *http.Response, responseBody interface{}) error {
```
Tem um comentário abaixo explicando melhor, mas acho que aqui não precisamos retornar um booleano